### PR TITLE
3496 - Geo UI refactor - Add nav Geo Sections component

### PR DESCRIPTION
**Pull request by yaguzmang**
Adding a GeoSection component inspired on the current NavigationSection component. And, adding a blank Background component (functionality coming next as it a lot of code and needs a PR of its own).

https://github.com/openforis/fra-platform/assets/41337901/9e5a3912-3c02-4b35-beef-9667f467a46d

The other sections were added just for the video.

### CODE REVIEW

File: src/client/components/Navigation/NavGeo/Sections/Background/Background.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+const Background: React.FC = () => {
+  return (
+    <div>
+      <span>Background content</span>
+    </div>
+  )
+}
+
+export default Background
```
Comment (yaguzmang) [inline src/client/components/Navigation/NavGeo/Sections/Background/Background.tsx]: I am thinking of adding the rest of the sections (forest, protected, and burned area layers) here under `src/client/components/Navigation/NavGeo/Sections/<Section name>`, but it's open for suggestions.

Comment (minotogna) [reply src/client/components/Navigation/NavGeo/Sections/Background/Background.tsx]: Add one peace at the time , as inner components of NavGeo

File: src/client/components/Navigation/NavGeo/NavGeo.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 import React from 'react'
 
+import GeoSection from 'client/components/Navigation/NavGeo/Sections'
+import Background from 'client/components/Navigation/NavGeo/Sections/Background'
+
 const NavGeo: React.FC = () => {
   return (
     <div className="nav-geo">
-      <h1>Nav Geo</h1>
+      <GeoSection key="background" showSections={false} sectionLabel="Background">
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/NavGeo.tsx]: Put each component as inner child , not in the Sections folder .
Plus the show sections can be handled within the GeoSection directly

File: src/client/components/Navigation/NavGeo/NavGeo.tsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
+import Background from 'client/components/Navigation/NavGeo/Background'
+import GeoSection from 'client/components/Navigation/NavGeo/Sections'
+
 const NavGeo: React.FC = () => {
   return (
     <div className="nav-geo">
-      <h1>Nav Geo</h1>
+      <GeoSection key="background" sectionLabel="Background">
+        <Background />
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/NavGeo.tsx]: Background is not needed. only 'Satellite mosaic '

Comment (minotogna) [reply src/client/components/Navigation/NavGeo/NavGeo.tsx]: Also @yaguzmang , let's start using i18n , only English for now . Ok ?

File: src/client/components/Navigation/NavGeo/Sections/GeoSection.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+const GeoSection: React.FC<Props> = (props) => {
+  const { children, sectionLabel } = props
+
+  const [expanded, setExpanded] = useState(false)
+
+  return (
+    <div>
+      <div
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/Sections/GeoSection.tsx]: I think we can use button here

File: src/client/components/Navigation/NavGeo/Sections/GeoSection.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+.nav-geo-section__items-hidden {
+  padding: 0 0 0 $spacing-m;
+  max-height: 0;
+  transition: max-height 0.15s ease-out;
+}
+
+.nav-geo-section__items-visible {
+  transition: max-height 0.25s ease-out;
+}
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/Sections/GeoSection.scss:31]: Are u using all these classes already? 
Add only what you need now and only now

File: src/client/components/Navigation/NavGeo/Sections/GeoSection.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import './GeoSection.scss'
+import React, { ReactElement, useState } from 'react'
+
+type Props = {
+  children: ReactElement
+  sectionLabel: string
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/Sections/GeoSection.tsx]: Section is already the name of the component . Call it simply label or labelKey if you plan to pass the i18n key

File: src/client/components/Navigation/NavGeo/NavGeo.tsx (hunk 3 above)
Comment (yaguzmang) [reply src/client/components/Navigation/NavGeo/NavGeo.tsx]: Sure. I added a new translations file `geo.js`

File: src/client/components/Navigation/NavGeo/Sections/GeoSection.scss (hunk 5 above)
Comment (yaguzmang) [reply src/client/components/Navigation/NavGeo/Sections/GeoSection.scss:31]: Yes, they are all used.

File: src/client/components/Navigation/NavGeo/NavGeo.tsx [hunk 7]
```diff
@@ -1,10 +1,17 @@
 import './NavGeo.scss'
 import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import SatelliteMosaic from 'client/components/Navigation/NavGeo/SatelliteMosaic'
+import GeoSection from 'client/components/Navigation/NavGeo/Sections'
```
Comment (minotogna) [inline src/client/components/Navigation/NavGeo/NavGeo.tsx:6]: In the next pr , please rename the folder to GeoSection , old name was Sections

**Review by minotogna — APPROVED** [review]

**Pull request merged**